### DISCUSSION TITLE
Can the Android StrongSwan client set local_ts and remote_ts?

### DISCUSSION THREAD

**Discussion by SunMengLong** [strongswan/strongswan#2164] [Q&A]
Hello, the server segment is not configured with rightsourceip, and an error message is reported on the Android end as follows：
received TS_UNACCEPTABLE notify, no CHILD_SA built
failed to establish CHILD_SA, keeping IKE_SA
received AUTH_LIFETIME of 10237s, scheduling reauthentication in 8798s
initiate failed: establishing CHILD_SA 'child' failed

However, Linux has configured both local_ts and remote_ts, and this error did not occur.

**Comment by tobiasbrunner** [comment]
Check the server log for the exact error. However, the Android client will always request a virtual IP, and if none is assigned by the server this won't work. This has nothing to do with the traffic selectors (which can't be configured on the client, it will always propose `0.0.0.0/0`).

**Reply by SunMengLong** [reply]
Server log:

Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> peer supports MOBIKE
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> authentication of '192.168.81.97' (myself) with pre-shared key
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> IKE_SA IKEV2-PSK[2] established between 192.168.81.97[192.168.81.97]...192.168.81.50[192.168.137.44]
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> scheduling reauthentication in 10079s
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> maximum IKE_SA lifetime 10619s
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> peer requested virtual IP 10.0.0.1
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> no virtual IP found for 10.0.0.1 requested by '192.168.137.44'
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> no virtual IP found, sending INTERNAL_ADDRESS_FAILURE
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> configuration payload negotiation failed, no CHILD_SA built
Mar 14 02:01:13 11[IKE] <IKEV2-PSK|2> failed to establish CHILD_SA, keeping IKE_SA
Mar 14 02:01:13 11[ENC] <IKEV2-PSK|2> generating IKE_AUTH response 1 [ IDr AUTH N(MOBIKE_SUP) N(ADD_4_ADDR) N(INT_ADDR_FAIL) ]
Mar 14 02:01:13 11[NET] <IKEV2-PSK|2> sending packet: from 192.168.81.97[4500] to 192.168.81.50[61832] (160 bytes)

**Reply by tobiasbrunner** [reply]
That's not a TS_UNACCEPTABLE notify like the first log shows, though, but an INTERNAL_ADDRESS_FAILURE notify (exactly because the client requested a virtual IP the server doesn't assign - however, this is not the Android client, as it seems to request a specific virtual IP, which the Android client doesn't do, and it also doesn't send a remote identity). In the first scenario some other error occurred.

**Reply by SunMengLong** [reply]
Can Android clients specify virtual IPs without being assigned by the server?

**Reply by tobiasbrunner** [reply]
No
